src/main/java/shef/servlets/UserServlet.java

```diff
@@ -101,27 +108,43 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     String bio = request.getParameter("bio-input");
 
     Key userKey = KeyFactory.createKey("User", id);
+    String keyString = KeyFactory.keyToString(userKey);
 
-    if (profilePictureUrl == null) {
-      try {
-        Entity originalEntity = datastore.get(userKey);
-        profilePictureUrl = (String) originalEntity.getProperty("profile-picture-url");
-      } catch (EntityNotFoundException e) {
-        throw new IOException("No profile picture provided.");
+    try {
+      // If updating an existing user, just update the changed fields.
+      Entity user = datastore.get(userKey);
+      if(email != null) {
+        user.setProperty("email", email);
+      }
+      if(username != null) {
+        user.setProperty("username", username);
+      }
+      if(location != null) {
+        user.setProperty("location", location);
+      }
+      if(profilePictureUrl != null) {
+        user.setProperty("profile-picture-url", profilePictureUrl);
+      }
+      if(bio != null) {
```

Comment:
Since there's a lot of repeated code here, is there a better way to do this? Maybe a method that takes in a property name and value as input and sets the property if it's not null, inside of a loop that calls it on each property?

Reply:
I thought about trying to do this as a loop, but I couldn't figure out how. Specifically, how would I get the first argument when setting the property?

Reply:
Actually, I could potentially have an array with the property names as well as an array with the properties themselves. As long as they're in the same order

Reply:
HttpServletRequests have the following method for getting a map of the parameters: https://docs.oracle.com/javaee/6/api/javax/servlet/ServletRequest.html#getParameterMap()

I was thinking that you could loop over the key-value pairs in this map, and then run the method on each pair.

Reply:
+1.

Reply:
I resolved it in a different, but that still removed the repeated code.